HabitTracker.GreyKlouds/HabitTracker/Program.cs

```diff
@@ -0,0 +1,248 @@
+                    closeApp = true;
+                    Environment.Exit(0);
+                    break;
+                default:
+                    Console.WriteLine("Invalid command. Press any key to start over");
+                    Console.ReadKey();
+                    Console.Clear();
+                    break;
+            }
+        }
+    }
+    private static void Insert()
+    {
+        string date = GetDateInput();
+
+        int quantity = GetNumberInput("\n\nPlease insert number of glasses or mesurment of your choice (Whole numbers only)\n\n");
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+            $"INSERT INTO DrinkingWater(date,quantity) VALUES('{date}',{quantity})";
+
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+        }
+    }
+    internal static void Update()
+    {
+        GetAllRecords();
+
+        var recordId = GetNumberInput("\n\nPlease type Id of the record would like to update. Type 0 to return to main manu.\n\n");
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+
+            var checkCmd = connection.CreateCommand();
+            checkCmd.CommandText = $"SELECT EXISTS(SELECT 1 FROM DrinkingWater WHERE Id = {recordId})";
+            int checkQuery = Convert.ToInt32(checkCmd.ExecuteScalar());
+
+            if (checkQuery == 0)
+            {
+                Console.WriteLine($"\n\nRecord with Id {recordId} doesn't exist.\n\n");
+                connection.Close();
+                Update();
+            }
+
+            string date = GetDateInput();
+
+            int quantity = GetNumberInput("\n\nPlease insert number of glasses or other measure of your choice (no decimals allowed)\n\n");
+
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText = $"UPDATE DrinkingWater SET date = '{date}', quantity = {quantity} WHERE Id = {recordId}";
+
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+        }
+
+
+    }
+    private static void GetAllRecords()
+    {
+        Console.Clear();
+        // create a connecton to the database:
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            //create a table command to show the data:
+            var tableCmd = connection.CreateCommand();
+
+            // SQL query:
+            tableCmd.CommandText =
+            @"select * from DrinkingWater";
+
+            //Create a list to display to the user:
+            List<DrinkingWater> tableData = new();
+            // read the SQL database:
+            SqliteDataReader reader = tableCmd.ExecuteReader();
+
+            Console.WriteLine("\t\tRecent Records:");
+            if (reader.HasRows)

```

Comment:
🟠 **Code Complexity**

💡 `if (reader.HasRows)` is a redundant level of indentation here, as `while (reader.Read())` will only be true if the reader has rows. Essentially, `reader.Read()` already does this check for you.

🔧 Remove the redundant `if` statement and only keep the `while` statement.